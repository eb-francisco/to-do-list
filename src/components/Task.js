import React, {Component} from 'react'
import {Form, Button} from 'react-bootstrap';

class Task extends Component {

    constructor(){
        super();

        this.state = {
            name: '',
            isCompleted: false,
        }
    }

    handleIsCompleted = () => {

        this.setState({
            isCompleted: !this.state.isCompleted
        })
    }

    render(){
        const {removeTask, task:{id}} = this.props;
    
        return(
            <div>
                <Form>
                    <Form.Group>
                        <Form.Label>Name</Form.Label>
                        <Form.Control 
                            className='input-name'
                            onChange={event => this.setState(
                                {name: event.target.value}
                            )}
                        />
                    </Form.Group>
                    <Form.Group>
                        <Form.Check label='Completed'
                            className='check-is-completed'
                            onChange={this.handleIsCompleted}
                        />
                    </Form.Group>
                    <Button className='btn-remove' onClick={() => removeTask(id)}>
                        Remove task
                    </Button>
                </Form>
            </div>
        )
    }
}

export default Task;
