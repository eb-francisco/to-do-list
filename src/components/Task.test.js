import React from 'react';
import {shallow} from 'enzyme';
import Task from './Task.js';


describe('Task', ()=>{
    const mockRemove = jest.fn();
    const id = 1;
    const props = { task:
        {id},
        removeTask: mockRemove,
    };
    const task = shallow(<Task {...props} />);

    it('should render correcly', () =>{
        expect(task.debug()).toMatchSnapshot();
    });

    it('intializes task name and is completed in state', () =>{
        expect(task.state()).toEqual({name:'', isCompleted: false})
    });

    describe('when typing on name field', () => {
        const toDoTask = 'hacer retro';

        beforeEach(()=>{
            task.find('.input-name').simulate('change', {target: {value: toDoTask}});
        })

        it('updates the name in state', () => {
            expect(task.state().name).toEqual(toDoTask);
        })
    });

    describe('when clicking on isCompleted', () => {
        beforeEach(() => {
            task.find('.check-is-completed').simulate('change')
        })

        it('updates is completed in state', () => {
            expect(task.state().isCompleted).toEqual(true);
        })
    })

    describe('when clicking on remove button', () => {
        beforeEach(() => {
            task.find('.btn-remove').simulate('click');
        });

        it('calls the remove function callback', () => {
            expect(mockRemove).toHaveBeenCalledWith(id);
        })
    });
});
