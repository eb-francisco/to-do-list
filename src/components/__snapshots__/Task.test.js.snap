// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task should render correcly 1`] = `
"<div>
  <Form inline={false}>
    <FormGroup>
      <FormLabel column={false} srOnly={false}>
        Name
      </FormLabel>
      <FormControl className=\\"input-name\\" onChange={[Function: onChange]} />
    </FormGroup>
    <FormGroup>
      <FormCheck label=\\"Completed\\" className=\\"check-is-completed\\" onChange={[Function]} type=\\"checkbox\\" inline={false} disabled={false} isValid={false} isInvalid={false} title=\\"\\" />
    </FormGroup>
    <Button className=\\"btn-remove\\" onClick={[Function: onClick]} variant=\\"primary\\" active={false} disabled={false} type=\\"button\\">
      Remove task
    </Button>
  </Form>
</div>"
`;
